<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>780</width>
    <height>543</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>171</width>
      <height>471</height>
     </rect>
    </property>
    <property name="title">
     <string>Input</string>
    </property>
    <widget class="QLabel" name="lblExamples">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>151</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string># Examples : 2000</string>
     </property>
    </widget>
    <widget class="QLabel" name="lblFeatures">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>151</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string># Features : 512</string>
     </property>
    </widget>
    <widget class="QLabel" name="lblExamples_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>120</y>
       <width>151</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Frequencies :</string>
     </property>
    </widget>
    <widget class="QTableWidget" name="tblFreq">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>140</y>
       <width>151</width>
       <height>131</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="btnLoad">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>151</width>
       <height>23</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Select the folder containing the data and the output file</string>
     </property>
     <property name="text">
      <string>Load Data</string>
     </property>
    </widget>
    <widget class="QLabel" name="lblClasses">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>151</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string># Classes : 56</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="cbResample">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>280</y>
       <width>141</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Resample rare classes</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="cbRegulate">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>300</y>
       <width>141</width>
       <height>17</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>make the # of emaples roughtly the same of all classes (1 standard div)</string>
     </property>
     <property name="text">
      <string>Regulate Frequencies</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_16">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>332</y>
       <width>71</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>TestSet % : </string>
     </property>
    </widget>
    <widget class="QSpinBox" name="sbBatchSize_2">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>330</y>
       <width>51</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>5</number>
     </property>
     <property name="maximum">
      <number>95</number>
     </property>
     <property name="value">
      <number>20</number>
     </property>
    </widget>
    <widget class="QSpinBox" name="sbBatchSize_3">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>358</y>
       <width>51</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>20000</number>
     </property>
     <property name="value">
      <number>1</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_17">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>360</y>
       <width>71</width>
       <height>16</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Any class having less examples than this won't be classified</string>
     </property>
     <property name="text">
      <string>Freq. Cutoff: </string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnSet">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>390</y>
       <width>131</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Set</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>10</y>
      <width>271</width>
      <height>471</height>
     </rect>
    </property>
    <property name="title">
     <string>Algorithm Configuration</string>
    </property>
    <widget class="QComboBox" name="cmbAlgorithm">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>20</y>
       <width>131</width>
       <height>22</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>Neural Network</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>SVM</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>K-NN</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>22</y>
       <width>101</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Algorithm :</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>50</y>
       <width>91</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Dimensions : &lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="txtDimensions">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>50</y>
       <width>131</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>512;256;56</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>80</y>
       <width>101</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Training by :</string>
     </property>
    </widget>
    <widget class="QComboBox" name="cmbTraining">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>80</y>
       <width>131</width>
       <height>22</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>Stochastic Gradient Descent</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Gradient Descent</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Online</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>110</y>
       <width>91</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Batch Size</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="sbBatchSize">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>110</y>
       <width>51</width>
       <height>22</height>
      </rect>
     </property>
     <property name="value">
      <number>10</number>
     </property>
    </widget>
    <widget class="QSpinBox" name="sbEpochs">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>140</y>
       <width>51</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>200</number>
     </property>
     <property name="value">
      <number>200</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>140</y>
       <width>91</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Number of Epochs</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>170</y>
       <width>91</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Learning Rate</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="txtLearningRat">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>170</y>
       <width>131</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>0.03</string>
     </property>
    </widget>
    <widget class="QPlainTextEdit" name="txtOutput">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>220</y>
       <width>251</width>
       <height>151</height>
      </rect>
     </property>
     <property name="plainText">
      <string>Batch 155160 ..
Batch 156/160 ..
Batch 157/160 ..
Batch 158/160 ..
Batch 159/160 ..
Batch 160/160 ..
Epoch 200/200 DONE. 
  (Correctly classified : 84%)

Time elapsed : 368 s
Training the NN on 1600 examples is finished.
Weights can be exported now.</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_15">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>207</y>
       <width>47</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>Output</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnTrain">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>380</y>
       <width>141</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Train !</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnStop">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>380</y>
       <width>61</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Stop</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnSet_2">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>430</y>
       <width>151</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Export Weights</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>10</y>
      <width>291</width>
      <height>471</height>
     </rect>
    </property>
    <property name="title">
     <string>Evaluation</string>
    </property>
    <widget class="QPushButton" name="pushButton">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>151</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Import Image ...</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="sbBatchSize_4">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>80</y>
       <width>51</width>
       <height>22</height>
      </rect>
     </property>
     <property name="value">
      <number>16</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>80</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Window Width</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnSet_3">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>20</y>
       <width>101</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Import Weights</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>100</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Window Height&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="sbBatchSize_5">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>100</y>
       <width>51</width>
       <height>22</height>
      </rect>
     </property>
     <property name="value">
      <number>16</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_11">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>120</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>H-stride</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="sbBatchSize_6">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>120</y>
       <width>51</width>
       <height>22</height>
      </rect>
     </property>
     <property name="value">
      <number>2</number>
     </property>
    </widget>
    <widget class="QSpinBox" name="sbBatchSize_7">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>140</y>
       <width>51</width>
       <height>22</height>
      </rect>
     </property>
     <property name="value">
      <number>2</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_12">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>140</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>V-stride</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_13">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>80</y>
       <width>71</width>
       <height>16</height>
      </rect>
     </property>
     <property name="layoutDirection">
      <enum>Qt::LeftToRight</enum>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">text-align: center;</string>
     </property>
     <property name="text">
      <string>Shrink Factor</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="txtLearningRat_2">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>100</y>
       <width>51</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>8</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_2">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>130</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Go !</string>
     </property>
    </widget>
    <widget class="QPlainTextEdit" name="txtOutput_2">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>370</y>
       <width>251</width>
       <height>91</height>
      </rect>
     </property>
     <property name="plainText">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>180</y>
       <width>251</width>
       <height>181</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border: 1px solid #aaa</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap>C:/Users/Ismail/Desktop/Page038.tif</pixmap>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>780</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
